enhancements/user-monitoring.md

```diff
@@ -0,0 +1,248 @@
+Currently, the cluster monitoring stack only reconciles OpenShift namespaces.
+
+## Motivation
+
+Currently there is no supported solution for customers to monitor their business critical applications.
+The existing cluster monitoring stack has no support for adding scrape targets in user namespaces
+as it is only available for OpenShift components scraping workloads from `openshift-*` namespaces.
+Other Red Hat products, outside the OpenShift framework components
+don’t have a monitoring solution to integrate their metrics and out-of-the-box insights into their well being.
+Currently it is difficult to achieve metering/chargeback for Red Hat products without a single source.
+
+### Goals
+
+Allow deployment and reconcilation scrape targets, recording rules, and alerting rules in user namespaces.
+This effectively allows the reconcilation of the following custom resources in user namespaces:
+
+- ServiceMonitor (https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#servicemonitor)
+- PodMonitor (https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#podmonitor)
+- PrometheusRule (https://github.com/openshift/prometheus-operator/blob/master/Documentation/api.md#prometheusrule)
+
+### Non-Goals
+
+Out of scope are:
+- Definition of Dashboards.
+- Deployment of Prometheus instances in user namespaces.
+- Deployment of Alertmanager instances user namespaces.
+
+## Proposal
+
+![](user-monitoring-deployment.png)
+
+The existing cluster monitoring stack is going to be extended with additional Prometheus (P) servers
+and one additional Prometheus Operator (PO) reconciling user namespaces
+as well as the newly added Prometheus servers.
+
+The cluster Prometheus Operator (PO) continues to operate the cluster-level Prometheus.
+We will continue to use a centralized Alertmanager (A) cluster.
+
+Querying data from a single, multi-tenant interface is done using the Thanos Querier (TQ) component.
+Soft tenancy is continued to be ensured by the existing topology leveraging kube-rbac-proxy [1]
+and prom-label-proxy [2].
+
+Differences between cluster- and user-level is mainly upgrade guarantees.
+As we keep cluster-level Prometheus immutable, we can guarantee a 100% cluster upgrade success
+but that rate may be lower for any additional Prometheus server scraping user namespaces.
+There are edge cases where an automated upgrade is not possible in the face of user monitoring.
+If users declare i.e. queries in ServiceMonitors which break in newer Prometheus versions
+we most likely cannot perform an automated migration, as PromQL is too complex to achieve that.
+
+Instead the "upgradable" condition in the ClusterOperator will be set to false.
+Routing configurations for alerts to be sent to specific teams need to be communicated/configured by an infrastructure owner.
+We will look into better mechanisms after GA.
```

Comment:
"better" is subjective, "extension mechanisms post GA." is appropriate I think

Reply:
replaced